java/src/main/java/org/lance/delta/DatasetDeltaBuilder.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.lance.delta;
+
+import org.lance.Dataset;
+import org.lance.JniLoader;
+
+import java.util.Optional;
+
+/**
+ * Builder for creating a {@link DatasetDelta} to explore changes between versions.
+ *
+ * <ul>
+ *   <li>Use comparedAgainstVersion to compare current dataset version.
+ *   <li>Or specify an explicit range with beginVersion and endVersion.
+ *   <li>These modes are mutually exclusive.
+ * </ul>
+ */
+public class DatasetDeltaBuilder {
+  static {
+    JniLoader.ensureLoaded();
+  }
+
+  private final Dataset dataset;
+  private Optional<Long> comparedAgainst = Optional.empty();
+  private Optional<Long> beginVersion = Optional.empty();
+  private Optional<Long> endVersion = Optional.empty();
+
+  public DatasetDeltaBuilder(Dataset dataset) {
+    this.dataset = dataset;
+  }
+
+  /** Shorthand to compare current dataset version against the given version. */
```

Comment:
Let me try to understand this:

1. set compared as the begin_version and the latest as the end version.
2. if comparedAgainst is set, the others would be ignored.

Am I right? I wonder if we can add some comments to explain this matter and do some validation in the java building phase. What do you think